js/framework/AnalyticsConstants.js

```diff
@@ -473,64 +544,123 @@ if (!OO.Analytics.EVENT_DATA)
+    var error = true;
     var toRet = data;
-    switch(expectedType)
+    for (var i = 0; i < expectedTypes.length; i++)
     {
-      case "string":
-        if (!toRet ||!OO._.isString(toRet))
+      var expectedType = expectedTypes[i];
+      if (expectedType === "string")
+      {
+        if (OO._.isString(toRet))
         {
-            error = true;
+          error = false;
+          break;
         }
-        break;
-      case "object":
-        if (!toRet || !OO._.isObject(toRet))
+      }
+      else if (expectedType === "object")
+      {
+        if (toRet && OO._.isObject(toRet))
         {
-            error = true;
+          error = false;
+          break;
         }
-      break;
-      case "number":
+      }
+      else if (expectedType === "number")
+      {
         // in the case number comes in as a string, try parsing it.
-        if (!OO._.isNumber(toRet))
+        var toRetFloat = parseFloat(toRet);
+        if (OO._.isNumber(toRet))
         {
-          toRet = parseFloat(toRet);
-          if (isNaN(toRet))
-          {
-            error = true;
-          }
+          error = false;
+          break;
         }
-        break;
-      case "boolean":
-        if (!OO._.isBoolean(toRet))
+        else if (!isNaN(toRetFloat))
         {
-          // consider string values "true" and "false" to be valid
-          if (toRet !== "true" || toRet !== "false")
-          {
-            error = true;
-          }
+          toRet = toRetFloat;
+          error = false;
+          break;
         }
-        break;
-      break;
+      }
+      else if (expectedType === "boolean")
+      {
+        if (OO._.isBoolean(toRet))
+        {
+          error = false;
+        }
+        else if (toRet === "true")
+        {
+          toRet = true;
+          error = false;
+          break;
+        }
+        else if (toRet === "false")
+        {
+          toRet = false;
+          error = false;
+          break;
+        }
+      }
     }
 
     if (error)
     {
-      OO.log("ERROR Analytics.EVENT_DATA." + className + " being created with invalid " + varName + ". Should be type \'" + expectedType + "\' but was \'" + typeof(data) + "\'.");
+      OO.log
+      (
+        "ERROR Analytics.EVENT_DATA." + className + " being created with invalid " + varName +
+        ". Should be one of these types [" + expectedTypes + "] but was [" + typeof(data) + "]."
+      );
       return undefined;
     }
 
     return toRet;
   };
 
+  /**
+   * @private
+   * @class Analytics.EVENT_DATA#checkAdType
+   * @classdesc Checks for a recognized Ad Type and returns the corresponding EVENT_DATA object.
+   * @property {string} adType The type of ad (linear video, linear overlay, nonlinear overlay)
+   * @property {object} adMetadata The metadata associated with the ad
+   * @returns {object} The EVENT_DATA object that associates with the Ad Type.
+   */
+  var checkAdType = function(adType, adMetadataIn)
+  {
+    var adMetadataOut;
```

Comment:
Can we return null and update the jsdoc to include a note about how this function can return null if no ad metadata is found for the ad type? We may also want to consider renaming the function to better fit the return value (the ad metadata).